src/models.rs

```diff
@@ -12,6 +16,83 @@ pub mod repos;
 pub mod teams;
 pub mod workflows;
 
+type BaseIdType = u64;
+
+macro_rules! id_type {
+    // This macro takes an argument of designator `ident` and
+    // creates a function named `$func_name`.
+    // The `ident` designator is used for variable/function names.
+    ($name:ident) => {
+        #[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+        pub struct $name(pub BaseIdType);
+        impl fmt::Display for $name {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                self.0.fmt(f)
+            }
+        }
+        impl Deref for $name {
+            type Target = BaseIdType;
+            fn deref(&self) -> &Self::Target {
+                &self.0
+            }
+        }
+        impl DerefMut for $name {
+            fn deref_mut(&mut self) -> &mut Self::Target {
+                &mut self.0
+            }
+        }
+        impl $name {
+            pub fn into_inner(self) -> BaseIdType {
+                self.0
+            }
+        }
+    };
+}
+
+id_type!(ActorId); // A Bot, EnterpriseUserAccount, Mannequin, Orangization or User
+id_type!(AppId);
+id_type!(AssetId);
+id_type!(CardId);
+id_type!(CheckRunId);
+id_type!(CommentId);
+id_type!(InstallationId);
+id_type!(IssueEventId);
+id_type!(IssueId);
+id_type!(JobId);
+id_type!(LabelId);
+id_type!(MilestoneId);
+id_type!(NotificationId);
+id_type!(OrgId);
+id_type!(ProjectId);
+id_type!(ProjectColumnId);
+id_type!(PullRequestId);
+id_type!(PushId);
+id_type!(ReleaseId);
+id_type!(RepositoryId);
+id_type!(ReviewId);
+id_type!(RunId);
+id_type!(StatusId);
+id_type!(TeamId);
+id_type!(UserId);
+id_type!(UserOrOrgId);
+id_type!(WorkflowId);
+
+macro_rules! convert_into {
+    ($from:ident, $to:ident) => {
```

Comment:
Ditto with this macro. I would probably have a syntax like `OrgId -> ActorId, UserId -> ActorId`. It'd look like this in the macro `($($from:ident -> $to:ident),+)`